src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -417,6 +397,34 @@ public int Seek(int char0, int char1, int char2)
             }
         }
 
+        private static int FindFirstEqualByte(Vector<byte> chEquals)
+        {
+            // Quasi-tree search
+            var vector64 = Vector.AsVectorInt64(chEquals);
+            for (var i = 0; i < Vector<long>.Count; i++)
+            {
+                var longValue = vector64[i];
+                if (longValue == 0) continue;
+
+                var shift = i << 1;
+                var offset = shift << 2;
+                var vector32 = Vector.AsVectorInt32(chEquals);
```

Comment:
Is probably worth a test; though the `AsVectorType` conversions shouldn't* do anything post Jit in release as its all the same register and should just be changing the cpu instructions called on them. Also one of the reasons I did it with vector indexers `[]` rather than shift+cast; is endianness confuses me.

*doesn't mean they don't